src/test/scala/cognite/spark/v1/SparkSchemaHelperTest.scala

```diff
@@ -84,13 +84,13 @@ class SparkSchemaHelperTest extends FlatSpec with ParallelTestExecution with Mat
   it should "construct Row from optional map type" in {
     val x = TestTypeOption(None, None, None, None, Some(Map("foo" -> "row", "bar" -> "a")), None, None, None)
     asRow(x) should
-      be(Row(None, None, None, None, Some(Map("foo" -> "row", "bar" -> "a")), None, None, None))
+      be(Row(null, null, null, null, Map("foo" -> "row", "bar" -> "a"), null, null, null))
   }
 
   it should "construct Row from optional seq type" in {
     val x = TestTypeOption(None, None, None, None, None, None, Some(Seq(None, Some(10L))), None)
     asRow(x) should
-      be(Row(None, None, None, None, None, None, Some(Seq(None, Some(10L))), None))
+      be(Row(null, null, null, null, null, null, Seq(None, Some(10L)), null))
```

Comment:
Just a curiosity question, I thought using `null`s was considered bad practice in Scala (here were also combining with a `Option`), I guess this is some limitation put on us by Spark?

Reply:
Correct! I'm a bit surprised that it worked on Spark 2.x tbh, because Spark seems to avoid Option in the public APIs, which I would guess is because they also have a Java API.